x-pack/solutions/observability/plugins/profiling/public/views/flamegraphs/flamegraph/index.tsx

```diff
@@ -55,6 +57,21 @@ export function FlameGraphView() {
     profilingRouter.push(routePath, { query: { ...query, searchText: newSearchText } });
   }
 
+  const { onPageReady } = usePerformanceContext();
+
+  if (state.status === AsyncStatus.Settled) {
+    onPageReady({
+      meta: {
+        rangeFrom,
+        rangeTo,
+      },
+      customMetrics: {
+        key1: 'totalSamples',
+        value1: state.data?.TotalSamples ?? 0,
+      },
+    });
+  }
```

Comment:
Let's wrap these calls to `onPageReady` in an `useEffect` to avoid calling this on every render